### ISSUE TITLE
Georgia-ReBORN 64 Bit Compatibility

### ISSUE THREAD

**Issue by TT-ReBORN** [TT-ReBORN/Georgia-ReBORN#102]
**All user-components Georgia-ReBORN is using need to be compiled in x64, 
otherwise you can not install the latest fb2k v2 x64, but you can use the fb2k v2 x86 version instead.**

The current state ( **08.10.2024** ) is as follows:
- foo_cuefixer ( optional, for iso images ) available in x64 :heavy_check_mark:
- foo_enhanced_playcount ( optional, for playcount ) available in x64 :heavy_check_mark:
- foo_musicbrainz ( optional, for tagging ) available in x64 :heavy_check_mark:
- foo_playcount ( optional, for playcount ) available in x64 :heavy_check_mark:
- foo_scrobble ( optional, for lastm.fm scrobbling ) available in x64 :heavy_check_mark:
- foo_spider_monkey_panel **( required, core )** not available in x64 :x:
- foo_ui_columns **( required, layout and settings )** available in x64 :heavy_check_mark:
- foo_ui_hacks **( required, hides toolbars and changes player sizes )** not available in x64 :x: 
( @TheQwertiest wants to implement this directly in SMP )
- foo_uie_eslyric ( optional, fetches lyrics ) available in x64 :heavy_check_mark:
- foo_vis_vumeter ( optional, makes peakmeter bar functional ) not available in x64 :x:
  ( There is a [new foo_vis_vumeter component in x64 by oops](https://hydrogenaud.io/index.php/topic,126733.0.html) but is not compatible with Columns UI and does not offer an ActiveX interface, this is needed for reading levels/peaks in realtime, **should be provided by SMP anyways**! )

We just need to wait for @TheQwertiest's magic, fingers crossed it will happen ;-)

-TT

**Comment by TheQwertiest** [comment]
PS: Regretfully SMP x64 wont be any time soon. Even in most optimistic scenario, the earliest I could continue development is mid-summer. And I don't think I will start working on ui_hacks features before I finish SMP v2 (which will take a lot of time)

**Comment by Sha1rholder** [comment]
Looking forward to 64bit version

**Comment by TT-ReBORN** [comment]
@Sha1rholder, 

unfortunately it is not in my hands and it depends when @TheQwertiest will release Spider Monkey Panel v2
which has x64 in his roadmap...

-TT

**Comment by Youxikong** [comment]
There is another panel called [jscript_panel](https://jscript-panel.github.io/) like Spider Monkey Panel. Maybe it can replace it.

**Comment by TT-ReBORN** [comment]
@Youxikong,

migrating Georgia-ReBORN to JScript Panel would represent a significant step backward, and I have no plans to pursue 
this option nor would I ever do that.

JScript Panel relies on very outdated JavaScript syntax and features, necessitating a complete rewrite of the entire theme, including all associated scripts. Furthermore, JScript Panel is confined to this antiquated version of JavaScript, and it is primarily used by very old scripts ( WinXP era), which are largely no longer being developed or maintained.

Spider Monkey Panel is the successor of JScript Panel, the dev -> @TheQwertiest only needs to be more active 
and finish Spider Monkey Panel v2.

-TT

**Comment by Youxikong** [comment]
Sorry, I've noticed JScript Panel use Windows Script Host which is outdated.🤐

**Comment by gabriels1963** [comment]
> @Youxikong,
> 
> migrating Georgia-ReBORN to JScript Panel would represent a significant step backward, and I have no plans to pursue this option nor would I ever do that.
> 
> JScript Panel relies on very outdated JavaScript syntax and features, necessitating a complete rewrite of the entire theme, including all associated scripts. Furthermore, JScript Panel is confined to this antiquated version of JavaScript, and it is primarily used by very old scripts (WinXP/7 era), which are largely no longer being developed or maintained.
> 
> Spider Monkey Panel is the successor of JScript Panel, the dev -> @TheQwertiest only needs to be more active and finish Spider Monkey Panel v2.
> 
> **In summary:**
> 
> * JScript Panel is limited to ES5 (ECMAScript 2009) only.
> * Spider Monkey Panel currently supports ES10 (ECMAScript 2019),
>   and it can be updated to newer ECMAScript versions to stay up-to-date.
> 
> -TT

I like SMP more compared to JScript Panel but it keeps crashing all the time on exit even on 32-bit version.
Too bad

**Comment by TT-ReBORN** [comment]
Can not confirm, it is stable and does NOT crash on exit. This is not an SMP issue but there is a known issue in the original Biography. In Gerogia-ReBORN this has been fixed for quite some time. I made a pull request here:
https://github.com/Wil-B/Biography/pulls
but it seems the original author @Wil-B is missing...

Other than that, SMP is only crashing ( out of memory ) if you have an extremely large music collection (300.000+ tracks) 
loaded when using foobar 32-bit because of 4 GB memory limitation. If we have foobar 64-bit with SMP 64-bit, there won't be an issues...

For more information, you can ask @TheQwertiest, the SMP dev...

-TT

**Comment by silverwings15** [comment]
@TT-ReBORN any progress on that x64 wrapper for SMP?

**Comment by TT-ReBORN** [comment]
Yes, Georgia-ReBORN is actually working with the JSplitter x64 but I need the UI Hacks and 
peakmeter audio monitoring functionalities in x64.

**Comment by silverwings15** [comment]
damn thats hardcore dedication, good luck

**Comment by TT-ReBORN** [comment]
**Important Update: As of 02 September 2025 and commit https://github.com/TT-ReBORN/Georgia-ReBORN/commit/501a9e85a4fe9179786e208713ac115dcdb488df, Georgia-ReBORN is now full converted to 64-bit. 
The versions currently available on the [Releases page](https://github.com/TT-ReBORN/Georgia-ReBORN/releases) are compatible only with foobar2000 32-bit. For the latest Georgia-ReBORN x64, download from the [master branch](https://github.com/TT-ReBORN/Georgia-ReBORN/archive/refs/heads/master.zip).**

---

Transition Georgia-ReBORN to 64-bit architecture
## Overview
This release transitions Georgia-ReBORN from 32-bit to 64-bit, removing unsupported components and introducing new ones for full compatibility, improved performance, and enhanced user experience.

## Component Changes
- **foo_spider_monkey_panel** (core component, unavailable in x64)
  - Removed and fully replaced by **foo_uie_jsplitter** for equivalent functionality.
- **foo_ui_hacks** (managed toolbar visibility and player sizing, unavailable in x64)
  - Removed and fully replaced by **foo_ui_wizard** for advanced UI management.
- **foo_vis_vumeter** (optional, enabled functional peakmeter bar, unavailable in x64)
  - Removed and fully replaced by **foo_audio_wizard** for peakmeter support.
- **Waveform Analysis Tools** (`audiowaveform` and `ffprobe`)
  - Removed and fully replaced by **foo_audio_wizard** for optimized waveform analysis.

## Enhancements
- **Improved Display Detection**: Powered by **foo_ui_wizard**, providing smoother and more reliable display handling.
- **Enhanced Maximize and Fullscreen Behavior**: Optimized via **foo_ui_wizard** for consistent and seamless UI interactions.
- **Removed UI-Hacks Workaround**: Eliminated the cosmetic 1px line fix, as **foo_ui_wizard** natively resolves this issue.
- **Faster Waveform Analysis**: Enabled by **foo_audio_wizard** for improved performance in waveform processing.

## Notes
- All removed components are no longer needed, as their functionality is fully provided by the new components in the 64-bit version.

-TT
